source/main/ipc/getHardwareWalletChannel.ts

```diff
@@ -477,56 +480,58 @@ export const handleHardwareWalletRequests = async (
       ? utils.str_to_path(stakingPathStr)
       : null;
 
-    try {
-      deviceConnection = get(devicesMemo, [devicePath, 'AdaConnection']);
+    deviceConnection = get(devicesMemo, [devicePath, 'AdaConnection']);
 
-      logger.info('[HW-DEBUG] DERIVE ADDRESS');
+    logger.info('[HW-DEBUG] DERIVE ADDRESS');
 
-      if (isTrezor) {
-        logger.info(
-          '[TREZOR-CONNECT] Called TrezorConnect.cardanoGetAddress()'
+    if (isTrezor) {
+      logger.info('[TREZOR-CONNECT] Called TrezorConnect.cardanoGetAddress()');
+
+      const result = await TrezorConnect.cardanoGetAddress({
+        showOnTrezor: true,
+        addressParameters: {
+          addressType,
+          path: `m/${spendingPathStr}`,
+          stakingPath: stakingPathStr ? `m/${stakingPathStr}` : null,
+        },
+        protocolMagic,
+        networkId,
+      });
+
+      if (result.success === false) {
+        logger.error(
+          '[TREZOR-CONNECT] TrezorConnect.cardanoGetAddress() failed',
+          result.payload
         );
 
-        const result = await TrezorConnect.cardanoGetAddress({
-          showOnTrezor: true,
-          device: buildTrezorDeviceParams(devicePath),
```

Comment:
I removed the functionality that defines Trezor path explicitly. When this parameter is provided, from 8.2.8, Trezor asks for passphrase on every call (which wasn't the case on 8.2.4), and making transaction internally makes two calls: first to get the public key (to ensure the right device is used), and second one to actually sign the transaction.

With the path parameter specified, user was asked for passphrase twice, which introduced room for user error (as different passphrases could be provided by mistake), without the parameter specified, TrezorConnect takes care of path management and everything works just fine.